driver/chain_syncer/chain_syncer_test.go

```diff
@@ -48,13 +74,76 @@ func (s *ChainSyncerTestSuite) TestSync() {
 
 // func (s *ChainSyncerTestSuite) TestSyncTriggerBeaconSync() {
 // 	s.s.p2pSyncVerifiedBlocks = true
+//  NOTE: need to increase the verified block as one of the conditions to trigger
+//         needBeaconSyncTriggered()
 // 	s.s.state.setLatestVerifiedBlockHash(common.Hash{})
 // }
 
+func (s *ChainSyncerTestSuite) TestAheadOfProtocolVerifiedHead2() {
+	s.TakeSnapshot()
+	// propose a couple blocks
+	testutils.ProposeAndInsertEmptyBlocks(&s.ClientTestSuite, s.p, s.s.calldataSyncer)
+
+	// NOTE: need to prove the proposed blocks to be verified, writing helper function
+
+	// generate transactopts to interact with TaikoL1 contract with.
+	privKey, err := crypto.ToECDSA(common.Hex2Bytes(os.Getenv("L1_PROVER_PRIVATE_KEY")))
+	s.Nil(err)
+	opts, err := bind.NewKeyedTransactorWithChainID(privKey, s.RpcClient.L1ChainID)
+	s.Nil(err)
+
+	head, err := s.RpcClient.L1.HeaderByNumber(context.Background(), nil)
+	s.Nil(err)
+
+	l2Head, err := s.RpcClient.L2.HeaderByNumber(context.Background(), nil)
+	s.Nil(err)
+
+	fmt.Printf("L1HeaderByNumber head: %v\n", head.Number)
```

Comment:
Lets use `log.Info` instead of `fmt.Printf` here, like other part of this codebase.